### ISSUE TITLE
Add Context IRIs

### ISSUE THREAD

**Issue by AtesComp** [AtesComp/rdf-transform#22]
Context IRIs are needed for supported quad export formats:
1. Add an option for a general Context IRI to generate quads or graphs.
2. Add a optional "graph" wrapper around root IRIs to override the general Context IRI.
3. Add an option to extend a given triple with a specific Context IRI to override the general and graph Context IRI.

**Comment by AtesComp** [comment]
# For Version 2.3.0
Added initial Context (i.e., Graph) IRI for preview and exports. This initial work uses the BaseIRI to create a Context IRI for the graph (there is currently only one graph per RDF Transform configuration). Future work should include a per ```root node``` Context IRI (like Type) and, later, possibly expanding to all Subject nodes

The code replaces the ```org.apache.jena.rdf.model.Model, ...rdf.model.ModelFactory, and ...graph.Triple``` with ```org.apache.jena.sparql.core.DatasetGraph, ...DatasetGraphFactory, and ...Quad```.

- For Preview, the output replaces Turtle with TriG
- For Exports, the output is as specified, but now outputs proper quads for Quad supporting formats 

The Context IRI is extracted from the BaseIRI by removing any dangling path, query, or fragment character (/, ?, #) so that it is a fully expressed Graph IRI (not a prefix). Examples:
```
    http://localhost:3030/           -->  http://localhost:3030
    http://www.example.org/my/data#  -->  http://www.example.org/my/data
```

**Comment by Yueqiao12Zhang** [comment]
Hi @AtesComp , after the update, I tried applying the same RDF skeleton to the same graph and export the N-Triples format, but the size changed from 13MB to more than 18GB and yet not finishing the download. I think there might be a bug. I can provide you all the files if you need a reference.

**Comment by AtesComp** [comment]
Yeah, it shouldn't do that unless the original data was being truncated for some reason.

How big is your data file?
I should only need one data file.  Post it here if it is small enough.  To "add files", you may need to rename them--just add ".txt" to the file name if it doesn't take it as is.  Also, post the RDF Transform exported template (a JSON file).

What OS are you running OR?
What OR version?
If Linux, what Java version?
Did RDF Transform 2.3.1 work for you (dialog resizing)?

**Comment by Yueqiao12Zhang** [comment]
Hi,

> What OS are you running OR?
I'm running on Mac OS 15.1 (24B83).

> What OR version?
OpenRefine 3.8.0

> If Linux, what Java version?
openjdk 21.0.1 2023-10-17
OpenJDK Runtime Environment (build 21.0.1+12-29)
OpenJDK 64-Bit Server VM (build 21.0.1+12-29, mixed mode, sharing)

> Did RDF Transform 2.3.1 work for you (dialog resizing)?
Yes, that worked!

This is the RISM skeleton file:
[rism_ontology.json](https://github.com/user-attachments/files/19573479/rism_ontology.json)

My input data file is also a RDF ttl file, but github does not support it. How should I send it to you?

Also, my project aims to reduce the possible output file size, can you make including the graph name in the export file an option?

**Comment by AtesComp** [comment]
Copy it, change the name to name.ttl.txt, and upload the  "txt" file.

For triple formats, it should not be using the graph name, only for quad formats.

Upgrading OpenRefine to 3.9.2 may fix the issue.  A related problem could be the version of Jena included with OpenRefine.  Later Jena versions changed the RDF exports.

**Comment by Yueqiao12Zhang** [comment]
Fixed, thank you very much!

**Comment by AtesComp** [comment]
I'll update the RDF Transform 2.3.0 documentation to indicate OpenRefine 3.9+ is required.
